src/main/java/uk/gov/hmcts/reform/pip/data/management/errorhandling/GlobalExceptionHandler.java

```diff
@@ -74,4 +77,19 @@
         return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(exceptionResponse);
     }
 
+    @ExceptionHandler(BlobStorageException.class)
+    public ResponseEntity<ExceptionResponse> handle(BlobStorageException ex) {
+        ExceptionResponse exceptionResponse = new ExceptionResponse();
+        if (Objects.equals(ex.getErrorCode().toString(), "BlobNotFound")) {
+            exceptionResponse.setMessage("404: Unable to find a blob matching the given inputs");
```

Comment:
I would avoid relying on Strings to determine which route to go down (as this would not be picked up at compilation time if Azure were to change the string in the future). 

This should probably be two different Exceptions, captured and then thrown at the source of where they are occurring.

Reply:
Solved :)